dask/array/reductions.py

```diff
@@ -223,7 +223,7 @@ def partial_reduce(func, x, split_every, keepdims=False, dtype=None, name=None):
         g = lol_tuples((x.name,), range(x.ndim), decided, dummy)
         dsk[(name,) + k] = (func, g)
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=[x])
-    return Array(graph, name, out_chunks, dtype=dtype)
+    return Array(graph, name, out_chunks, meta=x._meta.astype(dtype))
```

Comment:
I think that ideally we would call 

    func(x._meta, keepdims=keepdims)

That way if the shape or type of the array changes we'll learn about it.  Changing the dtype alone probably isn't enough.

Reply:
Good catch, I'll try to do that.

Reply:
It ended up being much more complicated to handle reductions, but just like slicing, the commit that I pushed to handle this case, seems to handle all cases I've seen in existing tests.